savoirfairelinux/sous-chef
# Route sheets report as PDF

## Fixes #696.

### Changes proposed in this pull request:
- delivery/views.py :
  - class RoutesInformation : generate and download pdf route sheets, create context only when displaying list of delivery routes, removed route sheet rendering as html for printing.
  - defineStyles() : refactored all reportlab paragraph styles.
  - class MultiRouteReport : generate pdf containing multiple route sheets.
  - drs_make_lines() : removed date because it was not used, ensured translation of component_group name.
- order/models.py : get_delivery_list() : ensured translation of component_group name.
- delivery/tests.py : adjusted tests for route sheets pdf printing to use attributes in response header instead of context, added test for multiple pages route sheets.
- delivery/fixtures/sample_data.json : added long delivery notes to test multiple pages route sheets.
- delivery/templates/routes_print.html : removed file because we now create a pdf instead.
- delivery/templates/routes.html : changed icon and tooltip to "download".

### Status

### How to verify this change
- The goal is to verify the expected behavior of issue #696.
- Make sure that a route has enough orders today and long delivery notes such that it will require more than one page to print.
- Make sure that a route needs an odd number of pages to print such that a blank page will be inserted on the back side of the last sheet.
- Click on Kitchen count
- Generate orders for the day
- Click on step 4-Routes
- Click confirm and then save for each of the routes; the button label will be "organized"
- Click on the Route Sheets button; this will download a pdf file
- Open the pdf file and verify the contents and formatting, footer text and page skips.
